superbench/runner/ansible.py

```diff
@@ -59,27 +61,34 @@ def __init__(self, config):
                 self._config['cmdline'] += ' --ask-pass --ask-become-pass'
         logger.info(self._config)
 
-    def run(self, ansible_config, sudo=False):    # pragma: no cover
+    def run(self, ansible_config, sudo=False, stdout=False):    # pragma: no cover
         """Run Ansible runner.
 
         Args:
             ansible_config (dict): Ansible config dict.
             sudo (bool): Run as sudo or not. Defaults to False.
+            stdout (bool): return stdout or not. Defaults to False.
 
         Returns:
             int: Ansible return code.
+            outputs: Ansible stdout if stdout is True.
         """
         if sudo:
             logger.info('Run as sudo ...')
             ansible_config['cmdline'] += ' --become'
         with tempfile.TemporaryDirectory(prefix='ansible') as tmpdir:
             r = ansible_runner.run(private_data_dir=tmpdir, **ansible_config)
             logger.debug(r.stats)
+            raw_outputs = r.stdout.read()
         if r.rc == 0:
             logger.info('Run succeed, return code {}.'.format(r.rc))
         else:
             self.failure_count += 1
             logger.warning('Run failed, return code {}.'.format(r.rc))
+        if stdout:
+            # Filter the color info of raw_outputs
+            outputs = re.sub(r'\x1b(\[.*?[@-~]|\].*?(\x07|\x1b\\))', '', raw_outputs)
```

Comment:
Could you help me understand what we are filtering here?

Reply:
`stdout` is a file storing the actual stdout as it appears in console which contains the ANSI escape color info.
This regular expression is used for ANSI standardized control sequences.
Reference: 
1. https://ansible-runner.readthedocs.io/en/stable/intro/#inputdir
2. https://en.wikipedia.org/wiki/ANSI_escape_code#CSI_codes